src/Controller/AdminController.php

```diff
@@ -0,0 +1,191 @@
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function addRoleBloggerAction(Request $request, User $id)
+    {
+
+        $em = $this->getDoctrine()->getManager();
+        $user = $em->getRepository(User::class)->find($id);
+        $em->persist($user);
+
+        $user->setRoles(['ROLE_BLOGGER', 'ROLE_USER']);
+
+        $em->flush();
+
+        return $this->redirectToRoute('admin');
+    }
+
+    /**
+     * @param Request $request
+     * @param User $id
+     * @Route("/admin/edit-user/role-admin/{id}", name="role_admin")
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function addRoleAdminAction(Request $request, User $id)
+    {
+
+        $em = $this->getDoctrine()->getManager();
+        $user = $em->getRepository(User::class)->find($id);
+        $em->persist($user);
+
+        $user->setRoles(['ROLE_ADMIN', 'ROLE_USER']);
+
+        $em->flush();
+
+        return $this->redirectToRoute('admin');
+    }
+
+
+    /**
+     * @param Request $request
+     * @Route("/admin/add-article", methods={"GET", "POST"}, name="add-article")
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function registerAction(Request $request)
+    {
+
+        $article = new Article();
+
+        $form = $this->createForm(AddArticleType::class, $article);
+
+        $firstName = $this->getUser()->getFirstName();
+
+        $lastName = $this->getUser()->getLastName();
+
+        $userId = $this->getUser()->getId();
+
+        if ($request->getMethod() == Request::METHOD_POST) {
+
+            $form->handleRequest($request);
+
+            $em = $this->getDoctrine()->getManager();
+
+            $em->persist($article);
+
+            $article->setAuthor($firstName . ' ' . $lastName);
+
+            $article->setAuthorId($userId);
+
+            $em->flush();
+
+            return $this->redirectToRoute('admin');
+        }
+
+        return $this->render('admin/addArticle.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/admin/edit-article", name="edit-article")
+     */
+    public function editArticleAction()
+    {
+        $repository = $this->getDoctrine()->getRepository(Article::class);
+
+        $articles = $repository->findAll();
+
+        return $this->render('admin/editArticle.html.twig', [
+            'articles' => $articles,
+        ]);
+    }
+
+    /**
+     * @param Request $request
+     * @param Article $id
+     * @Route("/admin/edit-article/delete/{id}", name="delete_article")
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
+     */
+    public function deleteArticleAction(Request $request, Article $id)
+    {
+
+        $em = $this->getDoctrine()->getManager();
+        $product = $em->getRepository(Article::class)->find($id);
```

Comment:
product? Try to use debugger, and look ad $id variable. Article is already there